src/test/java/com/metamx/emitter/core/EmitterTest.java

```diff
@@ -435,6 +450,67 @@ public void testBatchSplitting() throws Exception
     Assert.assertTrue(httpClient.succeeded());
   }
 
+  @Test
+  public void testBatchCompression() throws Exception
+  {
+    final List<UnitEvent> events = Arrays.asList(
+        new UnitEvent("plain-text", 1),
+        new UnitEvent("plain-text", 2)
+    );
+
+    emitter = sizeBasedEmitterWithCompression(2);
+
+    httpClient.setGoHandler(
+        new GoHandler()
+        {
+          @Override
+          public <Intermediate, Final> ListenableFuture<Final> go(Request request, HttpResponseHandler<Intermediate, Final> handler, Duration requestReadTimeout) throws Exception
+          {
+            Assert.assertEquals(new URL(TARGET_URL), request.getUrl());
+            Assert.assertEquals(
+                ImmutableList.of("application/json"),
+                request.getHeaders().get(HttpHeaders.Names.CONTENT_TYPE)
+            );
+            Assert.assertEquals(
+                ImmutableList.of(HttpHeaders.Values.GZIP),
+                request.getHeaders().get(HttpHeaders.Names.CONTENT_ENCODING)
+            );
+
+            byte[] bytes = request.getContent().array();
+            GZIPInputStream gis = new GZIPInputStream(new ByteArrayInputStream(bytes));
```

Comment:
I don't think its needed in tests, but FYI: `com.metamx.common.CompressionUtils#gzipInputStream`